src/App.tsx

```diff
@@ -2,108 +2,32 @@ import React from 'react';
 
 import './App.scss';
 
-// import postsFromServer from './api/posts';
-// import commentsFromServer from './api/comments';
-// import usersFromServer from './api/users';
+import { PostList } from './components/PostList';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
-export const App: React.FC = () => (
-  <section className="App">
-    <h1 className="App__title">Static list of posts</h1>
-
-    <div className="PostList">
-      <div className="PostInfo">
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">qui est esse</h3>
-
-          <p>
-            {' Posted by  '}
-
-            <a className="UserInfo" href="mailto:Sincere@april.biz">
-              Leanne Graham
-            </a>
-          </p>
-        </div>
-
-        <p className="PostInfo__body">
-          est rerum tempore vitae sequi sint nihil reprehenderit dolor beatae
-          ea dolores neque fugiat blanditiis voluptate porro vel nihil
-          molestiae ut reiciendis qui aperiam non debitis possimus qui neque
-          nisi nulla
-        </p>
-
-        <hr />
-
-        <b data-cy="NoCommentsMessage">No comments yet</b>
-      </div>
-
-      <div className="PostInfo">
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">
-            doloremque illum aliquid sunt
-          </h3>
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
 
-          <p>
-            {' Posted by  '}
+function getComments(postId: number): Comment[] | null {
+  const comments = commentsFromServer.filter(comment => (
+    comment.postId === postId
+  ));
 
-            <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-              Patricia Lebsack
-            </a>
-          </p>
-        </div>
+  return comments.length > 0 ? comments : null;
+}
```

Comment:
```suggestion
function getComments(postId: number): Comment[] {
  return commentsFromServer.filter(comment => (
    comment.postId === postId
  ));
}
```

I have doubts about this function work. You've decided to return null if no posts found. But this decision created a new type checking branch. What if I propose you to return only one type of data - `Comment[]`. Array filter method in any case will return you an array, so you don't need to worry about data type change.

So you can remove null from everywhere and check only if `comments.length` convertes to `true`.

What do you think?

Reply:
sounds like a way to get rid of unnecessary written code